key.go

```diff
@@ -34,37 +34,37 @@ import (
 )
 
 // LoadKey loads the key material based on the config
-func LoadKey(ctx context.Context, cfg HasKeyConfig, vault HasVaultConfig, which string) ([]byte, error) {
+func LoadKey(ctx context.Context, cfg KeyConfig, vault VaultConfig, which string) ([]byte, error) {
 	ctx, span := tracer.Start(ctx, "LoadKey")
 	defer span.End()
 
 	// If the key is not enabled, return an empty byte array
-	if en, ok := cfg.(HasEnabled); ok && !en.GetEnabled() {
+	if cfg.GetEnabled() {
 		return nil, nil
 	}
 
 	fields := logrus.Fields{
 		"key":           which,
-		"vault.enabled": vault != nil && vault.GetEnabled(),
+		"vault.enabled": vault.GetEnabled(),
 	}
 
 	method := "none"
 
-	if len(cfg.GetFile()) > 0 {
+	if len(cfg.File) > 0 {
 		method = "file"
-		fields["key.file"] = cfg.GetFile()
-		span.SetAttributes(attribute.String("key.file", cfg.GetFile()))
+		fields["key.file"] = cfg.File
+		span.SetAttributes(attribute.String("key.file", cfg.File))
 	}
 
-	if len(cfg.GetID()) > 0 {
-		if vault != nil && vault.GetEnabled() {
+	if len(cfg.ID) > 0 {
+		if vault.GetEnabled() {
```

Comment:
can vault be null here?

Reply:
Not anymore since we pass config by value instead of using the interface.